src/include/homestore/blkdata_service.hpp

```diff
@@ -139,6 +139,18 @@ class BlkDataService {
     folly::Future< std::error_code > async_read(MultiBlkId const& bid, sisl::sg_list& sgs, uint32_t size,
                                                 bool part_of_batch = false);
 
+    /**
+     * @brief Asynchronously reads data from the specified block ID.
+     *
+     * @param bids List of block IDs to read from.
+     * @param sgs The scatter-gather list to store the read data.
+     * @param part_of_batch Whether this read is part of a batch.
+     *
+     * @return A `folly::Future` that will contain the error code of the read operation.
+     */
+    folly::Future< std::error_code > async_read(std::vector<MultiBlkId> const& bids, sisl::sg_list& sgs, uint32_t size,
+                                                bool part_of_batch = false);
```

Comment:
I'm a little confused what this will be used for for a couple reasons:

1. There is only a single size; all all reads then assumed to be for the same length of data?
2. There is only a single Future provided, so all submitted READs have to pass/fail; this won't be useful to the target (SCSTConnector) since all READs are independent and get their own status (can succeed/fail independently). Is this purely for internal use?

Reply:
1. That is a good point. The size should not be a single variable
2. This API is for HomeBlocks internal use and scst connector will not use it. To be more specific, Each read() call that scst connector makes to Homeblocks will have lba and nlbas specified which is a contiguous lba range but the pbas are not necessarily contiguous. So, we need a vector of block ids to be read from solo repl dev